### PR TITLE
Fix #1192: HTTPSE shield setting is now scheme independent.

### LINKED ISSUE #1192: HTTPSE Setting Does Not Change Both HTTP/S Versions
### Description: 

Infinite loop when visiting digg.com with HTTPSE on. Unable to turn off to load website.
Related: https://github.com/brave/brave-ios/issues/688

### Steps to Reproduce 
  1. Turn off global HTTPSE
  1. Visit [http website](http://https-everywhere.badssl.com/)
  2. Turn on HTTPSE for current domain
  3. Site upgraded
  1. Observe that HTTPSE is _not_ set on the `https://` version

**Actual result:** 

HTTPSE setting is not scheme independent.

**Expected result:**

Changing HTTPSE setting for an `http` or `https` version of a website, should impact both `http` and `https` versions.

**Reproduces how often:** [Easily reproduced, Intermittent Issue]

**Brave Version:** 

1.9.3

**Device details:** 

**Website problems only:**
- did you check with Brave Shields down?
- did you check in Safari/Firefox (WkWebView-based browsers)?

### PR DESCRIPTION
**Pull request by jhreis**
## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-ios/issues) for my issue if one did not already exist.
- [x] My patch or PR title has a standard commit message that looks like `Fix #123: This fixes the shattered coffee cup!` (or `No Bug: <message>` if no relevant ticket)
- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`
- [x] New files have MPL-2.0 license header.

## Test Plan:

See https://github.com/brave/brave-ios/issues/1192
## Reviewer Checklist:

- [x] PR is linked to an issue via [Zenhub](https://www.zenhub.com/extension).
- [x] Issues are assigned to at least one epic.
- [x] Issues include necessary QA labels:
  - [x] `QA/(Yes|No)`
  - [x] `release-notes/(include|exclude)`
  - [x] `bug` / `enhancement`
- [x] Necessary security reviews have taken place.
- [x] Adequate test coverage exists to prevent regressions.
- [x] Adequate test plan exists for QA to validate (if applicable)

### CODE REVIEW

File: Client/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift [hunk 1]
```diff
@@ -160,6 +160,7 @@ extension BrowserViewController: WKNavigationDelegate {
                 if let mainDocumentURL = navigationAction.request.mainDocumentURL, url.scheme == "http" {
                     let domainForShields = Domain.getOrCreate(forUrl: mainDocumentURL)
                     let isPrivateBrowsing = PrivateBrowsingManager.shared.isPrivateBrowsing
+                    // Loading bad data
```
Comment (iccub) [inline Client/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift]: What does it mean?

**Review by iccub — CHANGES_REQUESTED** [review]
Code looks good, please add a test plan for QA sp they can verify it

File: Client/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift (hunk 1 above)
Comment (jhreis) [reply Client/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift]: Removed, shouldn't have been committed. Thanks for catching.

**Review by iccub — APPROVED** [review]

**Review by danishjafri88 — APPROVED** [review]
looks good

**Pull request merged**